PowerEditor/src/WinControls/ReadDirectoryChanges/ThreadSafeQueue.h

```diff
@@ -28,6 +28,70 @@
 
 #include <list>
 
+class CCriticalSection

```

Comment:
A very intuitive class, with its implementation quite similar to ATL's CComAutoCriticalSection. (Actually, there is nothing "Com" in it.)